File: apps/web/src/js/assessment.js
/* ============================================
   DebraWylde.world - Assessment Flow (Frontend Only)
   ============================================ */
(function () {
  "use strict";

  if (!document.body.classList.contains("assessment-page")) {
    return;
  }

  const STORAGE_KEY = "alignment_assessment_submission";
  const UNLOCK_KEY = "alignment_report_unlocked";

  const tieBreakPriority = [
    "Aligned Leadership™",
    "Decisive Expansion",
    "Vision & Architecture",
    "Identity Shift",
    "Strategic Disquiet"
  ];

  const questions = [
    { id: 1, group: "Strategic Disquiet", text: "My current role, business, or way of working no longer feels like a true fit." },
    { id: 2, group: "Strategic Disquiet", text: "I've built success, but something feels misaligned beneath the surface." },
    { id: 3, group: "Strategic Disquiet", text: "I can feel that a change is needed, even if I can't fully articulate it yet." },
    { id: 4, group: "Identity Shift", text: "I feel like I am outgrowing the role or identity I have been known for." },
    { id: 5, group: "Identity Shift", text: "I am releasing old expectations about who I should be." },
    { id: 6, group: "Identity Shift", text: "I sense that the version of me that created this chapter is not the one who will lead the next." },
    { id: 7, group: "Vision & Architecture", text: "I can feel a new direction emerging, but it is not yet fully defined." },
    { id: 8, group: "Vision & Architecture", text: "I need clearer structure around what I am building next." },
    { id: 9, group: "Vision & Architecture", text: "I am ready to translate what I sense into a more strategic vision." },
    { id: 10, group: "Decisive Expansion", text: "I know what needs to happen next, and I am ready to move with clarity." },
    { id: 11, group: "Decisive Expansion", text: "I am becoming more decisive about what stays and what no longer belongs." },
    { id: 12, group: "Decisive Expansion", text: "I am willing to expand into a bigger version of leadership." },
    { id: 13, group: "Aligned Leadership™", text: "My work, identity, and direction feel deeply integrated." },
    { id: 14, group: "Aligned Leadership™", text: "I am leading from a place of clarity rather than reaction." },
    { id: 15, group: "Aligned Leadership™", text: "I feel aligned with what I am building and how I am building it." }
  ];

  const answerScale = [
    { value: 1, label: "Not true for me" },
    { value: 2, label: "Rarely true" },
    { value: 3, label: "Sometimes true" },
    { value: 4, label: "Mostly true" },
    { value: 5, label: "Strongly true" }
  ];

  const resultsByStage = {
    "Strategic Disquiet": {
      headline: "Strategic Disquiet",
      subheadline: "Something no longer fits",
      body: "You've built success, but something in your current way of working no longer reflects who you are becoming. This stage is often subtle at first. Outwardly, everything may still appear functional. Internally, however, the misalignment is already present. This is not failure. It is the beginning of awareness.",
      requires: "Recognition, honesty, and space to acknowledge what is no longer aligned."
    },
    "Identity Shift": {
      headline: "Identity Shift",
      subheadline: "Releasing the old role",
      body: "You are no longer who you were when this chapter began. The familiar identity, role, or professional shape that once served you is starting to fall away. This stage can feel uncertain, but it is also necessary. Before your next direction becomes clear, an old version of self often has to be released.",
      requires: "Permission to let go, trust in transition, and support in redefining who you are becoming."
    },
    "Vision & Architecture": {
      headline: "Vision & Architecture",
      subheadline: "Designing what's next",
      body: "The shift is no longer only internal. You are beginning to sense a new direction and are ready to give it form. This is the point where insight must become structure. You do not need more noise. You need a clear architecture for what comes next.",
      requires: "Strategy, design, structure, and a clear framework for your next chapter."
    },
    "Decisive Expansion": {
      headline: "Decisive Expansion",
      subheadline: "Moving with clarity",
      body: "You are no longer waiting for certainty to appear. You are ready to move. This stage is about making clean decisions, choosing what matters, and stepping into a larger level of leadership with precision. Momentum comes from clarity, not force.",
      requires: "Decisive action, strategic refinement, and confident expansion."
    },
    "Aligned Leadership™": {
      headline: "Aligned Leadership™",
      subheadline: "Leading from alignment",
      body: "Your work, identity, and direction are becoming fully integrated. This stage is not about beginning again. It is about leading from coherence, depth, and strategic alignment. This is where your leadership becomes more powerful because it is no longer divided.",
      requires: "Sustained refinement, deeper embodiment, and expansion from an aligned core."
    }
  };

  function applyCRMTag(resultStage) {
    const map = {
      "Strategic Disquiet": "AA - Strategic Disquiet",
      "Identity Shift": "AA - Identity Shift",
      "Vision & Architecture": "AA - Vision & Architecture",
      "Decisive Expansion": "AA - Decisive Expansion",
      "Aligned Leadership™": "AA - Aligned Leadership"
    };
    const tag = map[resultStage] || "AA - Unassigned";
    console.log("[Assessment] CRM tag applied:", tag);
    return tag;
  }

  const state = {
    screen: "intro",
    currentQuestionIndex: 0,
    answersById: {},
    user: {
      first_name: "",
      email: ""
    },
    isUnlocked: false
  };

  const panelIntro = document.getElementById("panel-intro");
  const panelQuestions = document.getElementById("panel-questions");
  const panelLead = document.getElementById("panel-lead");
  const panelPaywall = document.getElementById("panel-paywall");
  const panelResults = document.getElementById("panel-results");

  const progressText = document.getElementById("progress-text");
  const progressBar = document.getElementById("progress-bar");
  const questionGroup = document.getElementById("question-group");
  const questionText = document.getElementById("question-text");
  const questionOptions = document.getElementById("question-options");

  const questionBack = document.getElementById("question-back");
  const questionNext = document.getElementById("question-next");
  const startBtn = document.getElementById("start-assessment");
  const leadForm = document.getElementById("lead-form");
  const leadBack = document.getElementById("lead-back");
  const paywallBack = document.getElementById("paywall-back");
  const unlockBtn = document.getElementById("unlock-report");

  const resultTitle = document.getElementById("result-title");
  const resultSubtitle = document.getElementById("result-subtitle");
  const resultBody = document.getElementById("result-body");
  const resultRequires = document.getElementById("result-requires");

  const firstNameInput = document.getElementById("lead-first-name");
  const emailInput = document.getElementById("lead-email");
  const websiteInput = document.getElementById("lead-website");
  const resultRequiresEl = document.getElementById("result-requires");
  const questionsSection = document.querySelector(".assessment-questions-section");

  let resultNoteEl = null;

  function setResultNote(message, isError) {
    if (!resultNoteEl) {
      resultNoteEl = document.createElement("p");
      resultNoteEl.className = "assessment-result-note";
      resultNoteEl.setAttribute("role", "status");
      if (resultRequiresEl && resultRequiresEl.parentNode) {
        resultRequiresEl.parentNode.insertBefore(
          resultNoteEl,
          resultRequiresEl.nextSibling
        );
      }
    }
    resultNoteEl.textContent = message;
    resultNoteEl.classList.toggle("assessment-result-note--error", !!isError);
    resultNoteEl.classList.toggle("is-visible", !!message);
  }

  function submitToBackend(resultStage, scores) {
    if (!window.DebraApi || typeof window.DebraApi.postJson !== "function") {
      return;
    }

    const answers = questions.map(function (question) {
      return {
        question_id: question.id,
        question_group: question.group,
        value: state.answersById[question.id] || 0
      };
    });

    const payload = {
      name: state.user.first_name,
      email: state.user.email,
      answers: answers,
      scores: scores,
      result_stage: resultStage,
      website: websiteInput ? websiteInput.value.trim() : "",
      source: "assessment",
      page: window.location.pathname
    };

    setResultNote("Sending your result summary by email...", false);

    window.DebraApi.postJson("/assessment", payload)
      .then(function () {
        setResultNote(
          "A copy of your result has been emailed to " + state.user.email + ".",
          false
        );
      })
      .catch(function () {
        // The on-screen result must survive even when the API is unavailable.
        setResultNote(
          "Your result is ready. We could not send the email summary right now, " +
            "but you can still review your result on this page.",
          true
        );
      });
  }

  function prefersReducedMotion() {
    return window.matchMedia("(prefers-reduced-motion: reduce)").matches;
  }

  function getScrollOffset() {
    const raw = getComputedStyle(document.documentElement).getPropertyValue("--site-header-height");
    const headerHeight = parseInt(raw, 10);
    return Number.isFinite(headerHeight) ? headerHeight : 72;
  }

  function scrollPanelIntoView(panel) {
    if (!panel) return;
    const offset = getScrollOffset() + 16;
    const top = panel.getBoundingClientRect().top + window.scrollY - offset;
    window.scrollTo({
      top: Math.max(0, top),
      behavior: prefersReducedMotion() ? "auto" : "smooth"
    });
  }

  function playPanelEntrance(panel) {
    if (!panel || prefersReducedMotion()) return;
    panel.classList.remove("is-panel-entering");
    void panel.offsetWidth;
    panel.classList.add("is-panel-entering");
    panel.addEventListener("animationend", function onEnd(event) {
      if (event.target !== panel) return;
      panel.classList.remove("is-panel-entering");
      panel.removeEventListener("animationend", onEnd);
    });
  }

  function playSectionRevealCue() {
    if (!questionsSection || prefersReducedMotion()) return;
    questionsSection.classList.remove("is-panel-revealed");
    void questionsSection.offsetWidth;
    questionsSection.classList.add("is-panel-revealed");
    questionsSection.addEventListener("animationend", function onEnd(event) {
      if (event.target !== questionsSection) return;
      questionsSection.classList.remove("is-panel-revealed");
      questionsSection.removeEventListener("animationend", onEnd);
    });
  }

  function revealQuestionsPanel() {
    requestAnimationFrame(function () {
      scrollPanelIntoView(panelQuestions);
      playPanelEntrance(panelQuestions);
      playSectionRevealCue();
    });
  }

  function showPanel(panelName) {
    const wasQuestionsHidden = panelQuestions.hidden;
    state.screen = panelName;
    if (panelIntro) {
      panelIntro.hidden = panelName !== "intro";
    }
    panelQuestions.hidden = panelName !== "questions";
    panelLead.hidden = panelName !== "lead";
    panelPaywall.hidden = panelName !== "paywall";
    panelResults.hidden = panelName !== "results";

    if (panelName === "questions" && wasQuestionsHidden) {
      revealQuestionsPanel();
    }
  }

  function selectedValueForCurrent() {
    const questionId = questions[state.currentQuestionIndex].id;
    return state.answersById[questionId];
  }

  function renderOptions(question) {
    questionOptions.innerHTML = "";
    const selectedValue = selectedValueForCurrent();

    answerScale.forEach(function (option) {
      const label = document.createElement("label");
      label.className = "assessment-option";
      if (selectedValue === option.value) {
        label.classList.add("is-selected");
      }

      const input = document.createElement("input");
      input.type = "radio";
      input.name = "assessment-answer";
      input.value = String(option.value);
      input.checked = selectedValue === option.value;
      input.addEventListener("change", function () {
        state.answersById[question.id] = Number(option.value);
        renderQuestion();
      });

      const text = document.createElement("span");
      text.textContent = option.label;

      label.appendChild(input);
      label.appendChild(text);
      questionOptions.appendChild(label);
    });
  }

  function renderQuestion() {
    const current = questions[state.currentQuestionIndex];
    const questionNo = state.currentQuestionIndex + 1;
    const pct = (questionNo / questions.length) * 100;
    const hasSelection = typeof selectedValueForCurrent() === "number";

    progressText.textContent = "Question " + questionNo + " of " + questions.length;
    progressBar.style.width = pct + "%";
    questionGroup.textContent = current.group;
    questionText.textContent = current.text;
    questionBack.disabled = state.currentQuestionIndex === 0;
    questionNext.disabled = !hasSelection;
    questionNext.textContent = questionNo === questions.length ? "Continue" : "Next";

    renderOptions(current);
  }

  function setFormError(input, isError) {
    const group = input.closest(".form-group");
    if (!group) return;
    if (isError) {
      group.classList.add("has-error");
    } else {
      group.classList.remove("has-error");
    }
  }

  function validateLeadForm() {
    const firstName = firstNameInput.value.trim();
    const email = emailInput.value.trim();
    const emailValid = /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email);

    setFormError(firstNameInput, !firstName);
    setFormError(emailInput, !email || !emailValid);

    if (!firstName) {
      firstNameInput.focus();
      return false;
    }
    if (!email || !emailValid) {
      emailInput.focus();
      return false;
    }

    state.user.first_name = firstName;
    state.user.email = email;
    return true;
  }

  function scoreGroups() {
    const scores = {
      "Strategic Disquiet": 0,
      "Identity Shift": 0,
      "Vision & Architecture": 0,
      "Decisive Expansion": 0,
      "Aligned Leadership™": 0
    };

    questions.forEach(function (question) {
      const val = state.answersById[question.id] || 0;
      scores[question.group] += val;
    });

    return scores;
  }

  function resolveResultStage(scores) {
    let best = tieBreakPriority[0];
    tieBreakPriority.forEach(function (stage) {
      if (scores[stage] > scores[best]) {
        best = stage;
      }
    });
    return best;
  }

  function buildSubmission(resultStage, scores) {
    const answers = questions.map(function (question) {
      return {
        question_id: question.id,
        question_group: question.group,
        value: state.answersById[question.id] || 0
      };
    });

    const crmTag = applyCRMTag(resultStage);

    const submission = {
      user: {
        first_name: state.user.first_name,
        email: state.user.email
      },
      answers: answers,
      scores: scores,
      result_stage: resultStage,
      crm_tag: crmTag,
      is_unlocked: state.isUnlocked,
      created_at: new Date().toISOString()
    };

    localStorage.setItem(STORAGE_KEY, JSON.stringify(submission));
    localStorage.setItem(UNLOCK_KEY, state.isUnlocked ? "true" : "false");

    return submission;
  }

  function renderResults() {
    const scores = scoreGroups();
    const resultStage = resolveResultStage(scores);
    const content = resultsByStage[resultStage];

    resultTitle.textContent = content.headline;
    resultSubtitle.textContent = content.subheadline;
    resultBody.textContent = content.body;
    resultRequires.textContent = "What this stage requires: " + content.requires;

    buildSubmission(resultStage, scores);

    showPanel("results");
    submitToBackend(resultStage, scores);
  }

  function goToQuestion(index) {
    if (index < 0) return;
    if (index >= questions.length) {
      showPanel("lead");
      return;
    }
    state.currentQuestionIndex = index;
    showPanel("questions");
    renderQuestion();
  }

  startBtn.addEventListener("click", function () {
    goToQuestion(0);
  });

  questionBack.addEventListener("click", function () {
    goToQuestion(state.currentQuestionIndex - 1);
  });

  questionNext.addEventListener("click", function () {
    if (questionNext.disabled) return;
    goToQuestion(state.currentQuestionIndex + 1);
  });

  leadBack.addEventListener("click", function () {
    goToQuestion(questions.length - 1);
  });

  leadForm.addEventListener("submit", function (event) {
    event.preventDefault();
    if (!validateLeadForm()) return;
    showPanel("paywall");
  });

  paywallBack.addEventListener("click", function () {
    showPanel("lead");
  });

  unlockBtn.addEventListener("click", function () {
    state.isUnlocked = true;
    localStorage.setItem(UNLOCK_KEY, "true");
    renderResults();
  });

  showPanel("intro");
})();

File: apps/api/app/routes/assessment.py
"""Alignment Assessment submission endpoint.

Scoring and stage content mirror the on-screen logic in
apps/web/src/js/assessment.js so the emailed summary matches what the user
sees. The server recomputes the stage from the submitted answers where possible,
and otherwise trusts the client-provided scores/stage as a fallback. The
assessment is always free and never gated behind payment.
"""

from html import escape
from typing import Dict, List, Optional

from fastapi import APIRouter, Request

from ..config import get_settings
from ..email_html import branded_email, email_button, email_detail_row
from ..email_service import EmailMessage, send_email
from ..models import insert_assessment, insert_lead
from ..schemas import AssessmentRequest, AssessmentSuccess
from ..security import check_honeypot, enforce_rate_limit, get_client_ip, hash_ip

router = APIRouter()

# Tie-break order: when scores are equal, the stage earlier in this list wins.
# This matches the frontend resolveResultStage logic.
TIE_BREAK_PRIORITY = [
    "Aligned Leadership\u2122",
    "Decisive Expansion",
    "Vision & Architecture",
    "Identity Shift",
    "Strategic Disquiet",
]

STAGE_CONTENT = {
    "Strategic Disquiet": {
        "subheadline": "Something no longer fits",
        "body": (
            "You have built success, but something in your current way of working "
            "no longer reflects who you are becoming. This stage is often subtle "
            "at first. Outwardly, everything may still appear functional. "
            "Internally, however, the misalignment is already present. This is not "
            "failure. It is the beginning of awareness."
        ),
        "requires": (
            "Recognition, honesty, and space to acknowledge what is no longer aligned."
        ),
    },
    "Identity Shift": {
        "subheadline": "Releasing the old role",
        "body": (
            "You are no longer who you were when this chapter began. The familiar "
            "identity, role, or professional shape that once served you is starting "
            "to fall away. This stage can feel uncertain, but it is also necessary. "
            "Before your next direction becomes clear, an old version of self often "
            "has to be released."
        ),
        "requires": (
            "Permission to let go, trust in transition, and support in redefining "
            "who you are becoming."
        ),
    },
    "Vision & Architecture": {
        "subheadline": "Designing what's next",
        "body": (
            "The shift is no longer only internal. You are beginning to sense a new "
            "direction and are ready to give it form. This is the point where "
            "insight must become structure. You do not need more noise. You need a "
            "clear architecture for what comes next."
        ),
        "requires": (
            "Strategy, design, structure, and a clear framework for your next chapter."
        ),
    },
    "Decisive Expansion": {
        "subheadline": "Moving with clarity",
        "body": (
            "You are no longer waiting for certainty to appear. You are ready to "
            "move. This stage is about making clean decisions, choosing what "
            "matters, and stepping into a larger level of leadership with "
            "precision. Momentum comes from clarity, not force."
        ),
        "requires": "Decisive action, strategic refinement, and confident expansion.",
    },
    "Aligned Leadership\u2122": {
        "subheadline": "Leading from alignment",
        "body": (
            "Your work, identity, and direction are becoming fully integrated. This "
            "stage is not about beginning again. It is about leading from coherence, "
            "depth, and strategic alignment. This is where your leadership becomes "
            "more powerful because it is no longer divided."
        ),
        "requires": (
            "Sustained refinement, deeper embodiment, and expansion from an aligned core."
        ),
    },
}

CRM_TAGS = {
    "Strategic Disquiet": "AA - Strategic Disquiet",
    "Identity Shift": "AA - Identity Shift",
    "Vision & Architecture": "AA - Vision & Architecture",
    "Decisive Expansion": "AA - Decisive Expansion",
    "Aligned Leadership\u2122": "AA - Aligned Leadership",
}

ASSESSMENT_INTRO = (
    "Thank you for completing the Alignment Assessment. "
    "Here is a summary of your result."
)


def _compute_scores(answers: List) -> Dict[str, float]:
    scores: Dict[str, float] = {stage: 0 for stage in TIE_BREAK_PRIORITY}
    for answer in answers:
        group = answer.question_group
        if group in scores:
            scores[group] += answer.value
    return scores


def _resolve_stage(scores: Dict[str, float]) -> str:
    best = TIE_BREAK_PRIORITY[0]
    for stage in TIE_BREAK_PRIORITY:
        if scores.get(stage, 0) > scores.get(best, 0):
            best = stage
    return best


@router.post("/assessment", response_model=AssessmentSuccess)
def submit_assessment(payload: AssessmentRequest, request: Request) -> AssessmentSuccess:
    check_honeypot(payload.website)
    enforce_rate_limit(request, "assessment")

    # Prefer server-side recomputation; fall back to client scores when no
    # itemised answers were supplied.
    if payload.answers:
        scores = _compute_scores(payload.answers)
        stage = _resolve_stage(scores)
    else:
        scores = {k: float(v) for k, v in (payload.scores or {}).items()}
        stage = payload.result_stage or (
            _resolve_stage(scores) if scores else TIE_BREAK_PRIORITY[0]
        )

    content = STAGE_CONTENT.get(stage)
    result_summary = content["body"] if content else ""
    stage_tag = CRM_TAGS.get(stage, "AA - Unassigned")

    answers_payload = [a.model_dump() for a in payload.answers]

    lead_id = insert_lead(
        source=payload.source or "assessment",
        name=payload.name,
        email=str(payload.email),
        subject="Alignment Assessment",
        stage_tag=stage_tag,
        consent_marketing=payload.consent_marketing,
        ip_hash=hash_ip(get_client_ip(request)),
        user_agent=request.headers.get("user-agent", "")[:300] or None,
        metadata={"page": payload.page, "result_stage": stage},
    )

    insert_assessment(
        lead_id=lead_id,
        name=payload.name,
        email=str(payload.email),
        answers=answers_payload,
        scores=scores,
        stage_tag=stage,
        result_summary=result_summary,
        metadata={"crm_tag": stage_tag, "page": payload.page},
    )

    _notify_internal(payload, stage, scores, lead_id)
    _email_result(payload, stage, content, lead_id)

    return AssessmentSuccess(
        message="Your result has been saved and a summary is on its way to your inbox.",
        result_stage=stage,
        result_summary=result_summary,
    )


def _site_page_url(base: str, page: str) -> str:
    root = (base or "").rstrip("/")
    path = page if page.startswith("/") else f"/{page}"
    return f"{root}{path}"


def _notify_internal(
    payload: AssessmentRequest, stage: str, scores: Dict[str, float], lead_id: int
) -> None:
    settings = get_settings()
    if not settings.internal_notification_email:
        return

    score_rows = "".join(
        "<tr>"
        f'<td style="padding:8px 0;font-size:14px;line-height:1.5;color:#B9AFC7;">'
        f"{escape(group)}</td>"
        f'<td align="right" style="padding:8px 0;font-size:15px;line-height:1.5;'
        f'font-weight:600;color:#E9CF7A;">{value}</td>'
        "</tr>"
        for group, value in scores.items()
    )
    email_link = (
        f'<a href="mailto:{escape(str(payload.email), quote=True)}" '
        f'style="color:#E9CF7A;text-decoration:none;">{escape(str(payload.email))}</a>'
    )
    detail_rows = "".join(
        [
            email_detail_row("Name", escape(payload.name)),
            email_detail_row("Email", email_link),
            email_detail_row("Result stage", escape(stage)),
            email_detail_row("Lead ID", f"#{lead_id}"),
        ]
    )
    inner = f"""
<p style="margin:0 0 10px 0;padding:0;font-size:11px;line-height:1.4;letter-spacing:0.26em;text-transform:uppercase;color:#D87C72;">Website Notification</p>
<h1 style="margin:0 0 22px 0;padding:0;font-family:Georgia,'Times New Roman',serif;font-size:30px;line-height:1.2;font-weight:400;color:#E9CF7A;">New assessment submission</h1>
<table role="presentation" border="0" cellpadding="0" cellspacing="0" width="100%" style="width:100%;background-color:#120526;border:1px solid #5a4570;border-radius:18px;">
  <tr>
    <td style="padding:22px 24px;">
      <p style="margin:0 0 14px 0;padding:0;font-size:12px;line-height:1.4;letter-spacing:0.18em;text-transform:uppercase;color:#C6A85A;">Lead details</p>
      <table role="presentation" border="0" cellpadding="0" cellspacing="0" width="100%">{detail_rows}</table>
    </td>
  </tr>
</table>
<table role="presentation" border="0" cellpadding="0" cellspacing="0" width="100%" style="margin-top:18px;width:100%;background-color:#391C59;border:1px solid #6b4568;border-radius:18px;">
  <tr>
    <td style="padding:22px 24px;">
      <p style="margin:0 0 12px 0;padding:0;font-size:12px;line-height:1.4;letter-spacing:0.18em;text-transform:uppercase;color:#D87C72;">Scores</p>
      <table role="presentation" border="0" cellpadding="0" cellspacing="0" width="100%">{score_rows}</table>
    </td>
  </tr>
</table>
"""
    html = branded_email(
        preheader=f"New assessment submission from {payload.name}: {stage}",
        inner_html=inner,
        footer_html=(
            '<p style="margin:0;padding:0;font-size:11px;line-height:1.6;color:#8F839F;">'
            "This notification was generated from the Debra Wylde Alignment Assessment."
            "</p>"
        ),
    )
    text = (
        f"New assessment submission (lead #{lead_id})\n"
        f"Name: {payload.name}\nEmail: {payload.email}\n"
        f"Result stage: {stage}\n"
        + "".join(f"  {g}: {v}\n" for g, v in scores.items())
    )
    send_email(
        EmailMessage(
            to_email=settings.internal_notification_email,
            subject=f"New assessment result: {stage} ({payload.name})",
            html=html,
            text=text,
            reply_to=str(payload.email),
            related_type="assessment",
            related_id=lead_id,
        )
    )


def _build_assessment_user_email(
    name: str,
    stage: str,
    content: Optional[dict],
    site_base_url: str,
) -> tuple[str, str]:
    """Build html and plain-text bodies for the user result summary email."""
    subheadline = content["subheadline"] if content else ""
    body = content["body"] if content else ""
    requires = content["requires"] if content else ""
    discovery_url = _site_page_url(site_base_url, "discovery-call.html")
    contact_url = _site_page_url(site_base_url, "contact.html")

    text = (
        f"Hi {name},\n\n"
        f"{ASSESSMENT_INTRO}\n\n"
        f"{stage}\n"
        f"{subheadline}\n\n"
        f"{body}\n\n"
        "What this stage requires\n"
        f"{requires}\n\n"
        "Next steps\n"
        f"Request a discovery call: {discovery_url}\n"
        f"Send a message: {contact_url}\n\n"
        "Warm regards,\n"
        "The Debra Wylde team\n"
    )
    inner = f"""
<p style="margin:0 0 18px 0;padding:0;font-size:16px;line-height:1.7;color:#F8F1E8;">Hi {escape(name)},</p>
<p style="margin:0 0 22px 0;padding:0;font-size:16px;line-height:1.7;color:#F8F1E8;">{escape(ASSESSMENT_INTRO)}</p>
<table role="presentation" border="0" cellpadding="0" cellspacing="0" width="100%" style="width:100%;background-color:#120526;border:1px solid #5a4570;border-radius:18px;">
  <tr>
    <td style="padding:24px;">
      <p style="margin:0 0 8px 0;padding:0;font-size:12px;line-height:1.4;letter-spacing:0.18em;text-transform:uppercase;color:#C6A85A;">Your result</p>
      <h1 style="margin:0 0 10px 0;padding:0;font-family:Georgia,'Times New Roman',serif;font-size:30px;line-height:1.2;font-weight:400;color:#E9CF7A;">{escape(stage)}</h1>
      <p style="margin:0;padding:0;font-size:15px;line-height:1.6;font-style:italic;color:#D87C72;">{escape(subheadline)}</p>
    </td>
  </tr>
</table>
<p style="margin:22px 0 18px 0;padding:0;font-size:16px;line-height:1.7;color:#F8F1E8;">{escape(body)}</p>
<table role="presentation" border="0" cellpadding="0" cellspacing="0" width="100%" style="width:100%;background-color:#391C59;border:1px solid #6b4568;border-radius:18px;">
  <tr>
    <td style="padding:22px 24px;">
      <p style="margin:0 0 10px 0;padding:0;font-size:12px;line-height:1.4;letter-spacing:0.18em;text-transform:uppercase;color:#D87C72;">What this stage requires</p>
      <p style="margin:0;padding:0;font-size:15px;line-height:1.65;color:#EDE2D2;">{escape(requires)}</p>
    </td>
  </tr>
</table>
<p style="margin:28px 0 14px 0;padding:0;font-size:12px;line-height:1.4;letter-spacing:0.18em;text-transform:uppercase;color:#C6A85A;">Next steps</p>
<p style="margin:0 0 18px 0;padding:0;font-size:15px;line-height:1.65;color:#EDE2D2;">If you would like to explore this result in conversation, book a discovery call or send Debra a message.</p>
<table role="presentation" border="0" cellpadding="0" cellspacing="0">
  <tr>
    <td style="padding:0;">
      {email_button(href=discovery_url, label="Discovery Call")}
    </td>
    <td style="padding:0;">
      {email_button(href=contact_url, label="Contact Debra")}
    </td>
  </tr>
</table>
<p style="margin:28px 0 0 0;padding:0;font-size:16px;line-height:1.7;color:#F8F1E8;">Warm regards,<br /><span style="color:#E9CF7A;">The Debra Wylde team</span></p>
"""
    html = branded_email(
        preheader=f"Your Alignment Assessment result: {stage}",
        inner_html=inner,
        footer_html=(
            '<p style="margin:0 0 8px 0;padding:0;font-size:12px;line-height:1.6;color:#B9AFC7;">'
            "Strategic counsel for high-achieving women leaders navigating expansion, "
            "transition, and their next level of impact."
            "</p>"
            '<p style="margin:0;padding:0;font-size:11px;line-height:1.6;color:#8F839F;">'
            "This email was sent because you completed the Alignment Assessment on the "
            "Debra Wylde website."
            "</p>"
        ),
    )
    return html, text


def _email_result(
    payload: AssessmentRequest,
    stage: str,
    content: Optional[dict],
    lead_id: int,
) -> None:
    settings = get_settings()
    html, text = _build_assessment_user_email(
        payload.name, stage, content, settings.site_base_url
    )
    send_email(
        EmailMessage(
            to_email=str(payload.email),
            subject=f"Your Alignment Assessment result: {stage}",
            html=html,
            text=text,
            related_type="assessment",
            related_id=lead_id,
        )
    )
